src/ApolloProvider.tsx

```diff
@@ -43,6 +43,14 @@ export default class ApolloProvider extends Component<ProviderProps, any> {
   constructor(props, context) {
     super(props, context);
 
+    this._init(props);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Also, we should probably put this in an `if` that checks if we actually have a new client.